packages/editor/src/epics/contractConfiguration/updateContractConfig.epic.ts

```diff
@@ -0,0 +1,53 @@
+import { ofType } from 'redux-observable';
+import { AnyAction } from 'redux';
+import { compilerActions, panesActions, contractConfigActions } from '../../actions';
+import { withLatestFrom, switchMap, catchError } from 'rxjs/operators';
+import { IProjectItem, ProjectItemTypes } from '../../models';
+import { findItemByPath } from '../../reducers/explorerLib';
+import { empty, of } from 'rxjs';
+
+export const updateContractConfig = (action$: AnyAction, state$: any) => action$.pipe(
+    ofType(compilerActions.HANDLE_COMPILE_OUTPUT),
+    withLatestFrom(state$),
+    switchMap(([action, state]) => {
+        console.log(state$);
+        const firstKey = Object.keys(action.data.contracts)[0];
+        const secondKey = Object.keys(action.data.contracts[firstKey])[0];
+        const compilerOutputData = action.data.contracts[firstKey][secondKey].metadata;
+        const parsedCompilerOutputData = JSON.parse(compilerOutputData);
+        const constructorData = parsedCompilerOutputData.output.abi.filter((obj: { type: string; }) => obj.type === 'constructor');
+        const numberOfArgs = constructorData[0].inputs.length;
```

Comment:
This will crash in case the `constructorData` is undefined, which could happen if there is no constructor in the contract (as it can be optional). Check out CryptoPizza template which doesn't have constructor

Reply:
I actually tested this, even if there is no constructor prior to updating the config, you will have to create the constructor anyway in case you want to add arguments to it and at point of saving a file it will already have the constructor in `constructorData` so I think this should be fine :)